frontend/src/views/instructor/ddahs/import.tsx

```diff
@@ -0,0 +1,319 @@
+    newDuty?: MinimalDuty;
+    status: DiffType;
+}
+
+function getDutiesDiff(oldDuties: RawDuty[], newDuties: RawDuty[]) {
+    let changes: DutyDiff[] = [];
+    let dutiesChanged = false;
+
+    oldDuties.forEach((oldDuty) => {
+        let oldDutyDeleted = true;
+
+        newDuties.forEach((newDuty, index) => {
+            if (
+                oldDuty.description === newDuty.description &&
+                oldDuty.hours === newDuty.hours
+            ) {
+                changes.push({ oldDuty, newDuty, status: DiffType.Unchanged });
+                newDuties.splice(index, 1);
+                oldDutyDeleted = false;
+            } else if (
+                oldDuty.description === newDuty.description &&
+                oldDuty.hours !== newDuty.hours
+            ) {
+                changes.push({ oldDuty, newDuty, status: DiffType.Updated });
+                newDuties.splice(index, 1);
+                oldDutyDeleted = false;
+                dutiesChanged = true;
+            }
+        });
+
+        // If we did not find old duties to match the new one - mark it as created
+        if (oldDutyDeleted) {
+            changes.push({ oldDuty, status: DiffType.Deleted });
+            dutiesChanged = true;
+        }
+    });
+
+    // The new duties that are left were just created - add them
+    newDuties.forEach((duty) =>
+        changes.push({ newDuty: duty, status: DiffType.Created })
+    );
+
+    // Do not return changes if everything is unchanged
+    return dutiesChanged ? changes : [];
+}
+
+function dutiesDiffList(changes: DutyDiff[]) {
+    return (
+        <ul>
+            {changes.map(({ oldDuty, newDuty, status }) => {
+                switch (status) {
+                    case DiffType.Unchanged:
+                        return (
+                            <li className="duty-unchanged">
+                                {oldDuty!.description} - {oldDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Updated:
+                        return (
+                            <li className="duty-updated">
+                                {oldDuty!.description} - {oldDuty!.hours}h{" "}
+                                <FaArrowRight />
+                                {newDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Created:
+                        return (
+                            <li className="duty-created">
+                                {newDuty!.description} - {newDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Deleted:
+                        return (
+                            <li className="duty-deleted">
+                                {oldDuty!.description} - {oldDuty!.hours}h
+                            </li>
+                        );
+                    default:
+                        return null;
```

Comment:
You should never get here, right? You should be able to do `null as never` or something and that will make typescript check that you never reach this point in the code.